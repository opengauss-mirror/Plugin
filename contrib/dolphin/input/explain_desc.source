create schema db_explain_desc;
set current_schema to 'db_explain_desc';

create table ed_t(c1 int, c2 varchar(100), c3 int default 10);
insert into ed_t values(generate_series(1, 10), 'hello', 100);

-- 1.use explain to query table's info
explain ed_t;
explain db_explain_desc.ed_t;

-- 2.use desc to query plan info
desc select c1, c2, c3 from ed_t;
desc insert into ed_t values(100, 'hello', 100);
desc update ed_t set c2 = 'world' where c1 < 5;
desc extended delete from ed_t where c1 < 5;
describe select c1, c2, c3 from ed_t;
describe insert into ed_t values(100, 'hello', 100);
describe extended update ed_t set c2 = 'world' where c1 < 5;
describe extended delete from ed_t where c1 < 5;
desc format=json select c1, c2, c3 from ed_t;
describe format=traditional insert into ed_t values(100, 'hello', 100);

-- 3.explain with key world: extended
explain extended select c1, c2, c3 from ed_t;
explain extended insert into ed_t values(100, 'hello', 100);
explain extended update ed_t set c2 = 'world' where c1 < 5;
explain extended delete from ed_t where c1 < 5;

-- 4.explain with specifying format(json or traditional)
explain format=json select c1, c2, c3 from ed_t;
explain format=json insert into ed_t values(100, 'hello', 100);
explain format=JSON update ed_t set c2 = 'world' where c1 < 5;
explain FORMAT=JSON update ed_t set c2 = 'world' where c1 < 5;
explain format=json delete from ed_t where c1 < 5;
explain format=Json delete from ed_t where c1 < 5;
explain format=JsOn delete from ed_t where c1 < 5;
explain format='JsOn' delete from ed_t where c1 < 5;
explain format='json' delete from ed_t where c1 < 5;
explain format=TRADITIONAL select c1, c2, c3 from ed_t;
explain FORMAT=TRADITIONAL select c1, c2, c3 from ed_t;
explain format=TRADITIONAL insert into ed_t values(100, 'hello', 100);
explain format=TRaDITiONAL update ed_t set c2 = 'world' where c1 < 5;
explain format=traditional delete from ed_t where c1 < 5;
explain format='traditional' delete from ed_t where c1 < 5;
explain format='TraDitional' delete from ed_t where c1 < 5;

drop table ed_t;

drop schema db_explain_desc cascade;
reset current_schema;

-- 5.use special word as table name
create table explain(c1 int, explain varchar(10));
create table describe(explain int, c2 varchar(10));
insert into explain values(1, 'explain');
insert into describe values(1, 'explain');
select * from explain;
select * from describe;

explain explain;
explain describe;
desc explain;
desc describe;
describe explain;
describe describe;

explain (costs off) select * from explain;
explain (costs off) select * from describe;
desc (costs off) select * from explain;
desc (costs off) select * from describe;
describe (costs off) select * from explain;
describe (costs off) select * from describe;

-- 6.test desc&explain with shell
\! gsql -d contrib_regression -r -p @portstring@ -c "describe explain"
\! gsql -d contrib_regression -r -p @portstring@ -c "explain describe"
\! gsql -d contrib_regression -r -p @portstring@ -c "desc explain"

\! gsql -d contrib_regression -r -p @portstring@ -c "show processlist" > /dev/null
\! gsql -d contrib_regression -r -p @portstring@ -c "show full processlist" > /dev/null

drop table explain, describe;
