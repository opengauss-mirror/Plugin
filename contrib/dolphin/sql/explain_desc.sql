create database db_explain_desc with dbcompatibility 'B';
\c db_explain_desc

create table ed_t(c1 int, c2 varchar(100), c3 int default 10);
insert into ed_t values(generate_series(1, 10), 'hello', 100);

-- 1.use explain to query table's info
explain ed_t;
explain public.ed_t;

-- 2.use desc to query plan info
desc select c1, c2, c3 from ed_t;
desc insert into ed_t values(100, 'hello', 100);
desc update ed_t set c2 = 'world' where c1 < 5;
desc extended delete from ed_t where c1 < 5;
describe select c1, c2, c3 from ed_t;
describe insert into ed_t values(100, 'hello', 100);
describe extended update ed_t set c2 = 'world' where c1 < 5;
describe extended delete from ed_t where c1 < 5;
desc format=json select c1, c2, c3 from ed_t;
describe format=traditional insert into ed_t values(100, 'hello', 100);

-- 3.explain with key world: extended
explain extended select c1, c2, c3 from ed_t;
explain extended insert into ed_t values(100, 'hello', 100);
explain extended update ed_t set c2 = 'world' where c1 < 5;
explain extended delete from ed_t where c1 < 5;

-- 4.explain with specifying format(json or traditional)
explain format=json select c1, c2, c3 from ed_t;
explain format=json insert into ed_t values(100, 'hello', 100);
explain format=JSON update ed_t set c2 = 'world' where c1 < 5;
explain format=json delete from ed_t where c1 < 5;
explain format=Json delete from ed_t where c1 < 5;
explain format=JsOn delete from ed_t where c1 < 5;
explain format='JsOn' delete from ed_t where c1 < 5;
explain format='json' delete from ed_t where c1 < 5;
explain format=TRADITIONAL select c1, c2, c3 from ed_t;
explain format=TRADITIONAL insert into ed_t values(100, 'hello', 100);
explain format=TRaDITiONAL update ed_t set c2 = 'world' where c1 < 5;
explain format=traditional delete from ed_t where c1 < 5;
explain format='traditional' delete from ed_t where c1 < 5;
explain format='TraDitional' delete from ed_t where c1 < 5;

drop table ed_t;

\c postgres

drop database db_explain_desc;
