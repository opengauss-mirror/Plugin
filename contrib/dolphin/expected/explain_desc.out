create schema db_explain_desc;
set current_schema to 'db_explain_desc';
create table ed_t(c1 int, c2 varchar(100), c3 int default 10);
insert into ed_t values(generate_series(1, 10), 'hello', 100);
-- 1.use explain to query table's info
explain ed_t;
 Field |          Type          | Null | Key | Default | Extra 
-------+------------------------+------+-----+---------+-------
 c1    | integer                | YES  |     | NULL    | 
 c2    | character varying(100) | YES  |     | NULL    | 
 c3    | integer                | YES  |     | 10      | 
(3 rows)

explain db_explain_desc.ed_t;
 Field |          Type          | Null | Key | Default | Extra 
-------+------------------------+------+-----+---------+-------
 c1    | integer                | YES  |     | NULL    | 
 c2    | character varying(100) | YES  |     | NULL    | 
 c3    | integer                | YES  |     | 10      | 
(3 rows)

-- 2.use desc to query plan info
desc select c1, c2, c3 from ed_t;
                       QUERY PLAN                        
---------------------------------------------------------
 Seq Scan on ed_t  (cost=0.00..13.19 rows=319 width=226)
(1 row)

desc insert into ed_t values(100, 'hello', 100);
                    QUERY PLAN                    
--------------------------------------------------
 [Bypass]
 Insert on ed_t  (cost=0.00..0.01 rows=1 width=0)
   ->  Result  (cost=0.00..0.01 rows=1 width=0)
(3 rows)

desc update ed_t set c2 = 'world' where c1 < 5;
                          QUERY PLAN                          
--------------------------------------------------------------
 Update on ed_t  (cost=0.00..13.99 rows=106 width=14)
   ->  Seq Scan on ed_t  (cost=0.00..13.99 rows=106 width=14)
         Filter: (c1 < 5)
(3 rows)

desc extended delete from ed_t where c1 < 5;
                         QUERY PLAN                          
-------------------------------------------------------------
 Delete on ed_t  (cost=0.00..13.99 rows=106 width=6)
   ->  Seq Scan on ed_t  (cost=0.00..13.99 rows=106 width=6)
         Filter: (c1 < 5)
(3 rows)

describe select c1, c2, c3 from ed_t;
                       QUERY PLAN                        
---------------------------------------------------------
 Seq Scan on ed_t  (cost=0.00..13.19 rows=319 width=226)
(1 row)

describe insert into ed_t values(100, 'hello', 100);
                    QUERY PLAN                    
--------------------------------------------------
 [Bypass]
 Insert on ed_t  (cost=0.00..0.01 rows=1 width=0)
   ->  Result  (cost=0.00..0.01 rows=1 width=0)
(3 rows)

describe extended update ed_t set c2 = 'world' where c1 < 5;
                          QUERY PLAN                          
--------------------------------------------------------------
 Update on ed_t  (cost=0.00..13.99 rows=106 width=14)
   ->  Seq Scan on ed_t  (cost=0.00..13.99 rows=106 width=14)
         Filter: (c1 < 5)
(3 rows)

describe extended delete from ed_t where c1 < 5;
                         QUERY PLAN                          
-------------------------------------------------------------
 Delete on ed_t  (cost=0.00..13.99 rows=106 width=6)
   ->  Seq Scan on ed_t  (cost=0.00..13.99 rows=106 width=6)
         Filter: (c1 < 5)
(3 rows)

desc format=json select c1, c2, c3 from ed_t;
           QUERY PLAN           
--------------------------------
 [                             +
   {                           +
     "Plan": {                 +
       "Node Type": "Seq Scan",+
       "Relation Name": "ed_t",+
       "Alias": "ed_t",        +
       "Startup Cost": 0.00,   +
       "Total Cost": 13.19,    +
       "Plan Rows": 319,       +
       "Plan Width": 226       +
     }                         +
   }                           +
 ]
(1 row)

describe format=traditional insert into ed_t values(100, 'hello', 100);
                    QUERY PLAN                    
--------------------------------------------------
 [Bypass]
 Insert on ed_t  (cost=0.00..0.01 rows=1 width=0)
   ->  Result  (cost=0.00..0.01 rows=1 width=0)
(3 rows)

-- 3.explain with key world: extended
explain extended select c1, c2, c3 from ed_t;
                       QUERY PLAN                        
---------------------------------------------------------
 Seq Scan on ed_t  (cost=0.00..13.19 rows=319 width=226)
(1 row)

explain extended insert into ed_t values(100, 'hello', 100);
                    QUERY PLAN                    
--------------------------------------------------
 [Bypass]
 Insert on ed_t  (cost=0.00..0.01 rows=1 width=0)
   ->  Result  (cost=0.00..0.01 rows=1 width=0)
(3 rows)

explain extended update ed_t set c2 = 'world' where c1 < 5;
                          QUERY PLAN                          
--------------------------------------------------------------
 Update on ed_t  (cost=0.00..13.99 rows=106 width=14)
   ->  Seq Scan on ed_t  (cost=0.00..13.99 rows=106 width=14)
         Filter: (c1 < 5)
(3 rows)

explain extended delete from ed_t where c1 < 5;
                         QUERY PLAN                          
-------------------------------------------------------------
 Delete on ed_t  (cost=0.00..13.99 rows=106 width=6)
   ->  Seq Scan on ed_t  (cost=0.00..13.99 rows=106 width=6)
         Filter: (c1 < 5)
(3 rows)

-- 4.explain with specifying format(json or traditional)
explain format=json select c1, c2, c3 from ed_t;
           QUERY PLAN           
--------------------------------
 [                             +
   {                           +
     "Plan": {                 +
       "Node Type": "Seq Scan",+
       "Relation Name": "ed_t",+
       "Alias": "ed_t",        +
       "Startup Cost": 0.00,   +
       "Total Cost": 13.19,    +
       "Plan Rows": 319,       +
       "Plan Width": 226       +
     }                         +
   }                           +
 ]
(1 row)

explain format=json insert into ed_t values(100, 'hello', 100);
                 QUERY PLAN                 
--------------------------------------------
 [                                         +
   {                                       +
     "Plan": {                             +
       "Node Type": "ModifyTable",         +
       "Operation": "Insert",              +
       "Relation Name": "ed_t",            +
       "Alias": "ed_t",                    +
       "Startup Cost": 0.00,               +
       "Total Cost": 0.01,                 +
       "Plan Rows": 1,                     +
       "Plan Width": 0,                    +
       "Plans": [                          +
         {                                 +
           "Node Type": "Result",          +
           "Parent Relationship": "Member",+
           "Startup Cost": 0.00,           +
           "Total Cost": 0.01,             +
           "Plan Rows": 1,                 +
           "Plan Width": 0                 +
         }                                 +
       ]                                   +
     }                                     +
   }                                       +
 ]
(1 row)

explain format=JSON update ed_t set c2 = 'world' where c1 < 5;
                 QUERY PLAN                 
--------------------------------------------
 [                                         +
   {                                       +
     "Plan": {                             +
       "Node Type": "ModifyTable",         +
       "Operation": "Update",              +
       "Relation Name": "ed_t",            +
       "Alias": "ed_t",                    +
       "Startup Cost": 0.00,               +
       "Total Cost": 13.99,                +
       "Plan Rows": 106,                   +
       "Plan Width": 14,                   +
       "Plans": [                          +
         {                                 +
           "Node Type": "Seq Scan",        +
           "Parent Relationship": "Member",+
           "Relation Name": "ed_t",        +
           "Alias": "ed_t",                +
           "Startup Cost": 0.00,           +
           "Total Cost": 13.99,            +
           "Plan Rows": 106,               +
           "Plan Width": 14,               +
           "Filter": "(c1 < 5)"            +
         }                                 +
       ]                                   +
     }                                     +
   }                                       +
 ]
(1 row)

explain format=json delete from ed_t where c1 < 5;
                 QUERY PLAN                 
--------------------------------------------
 [                                         +
   {                                       +
     "Plan": {                             +
       "Node Type": "ModifyTable",         +
       "Operation": "Delete",              +
       "Relation Name": "ed_t",            +
       "Alias": "ed_t",                    +
       "Startup Cost": 0.00,               +
       "Total Cost": 13.99,                +
       "Plan Rows": 106,                   +
       "Plan Width": 6,                    +
       "Plans": [                          +
         {                                 +
           "Node Type": "Seq Scan",        +
           "Parent Relationship": "Member",+
           "Relation Name": "ed_t",        +
           "Alias": "ed_t",                +
           "Startup Cost": 0.00,           +
           "Total Cost": 13.99,            +
           "Plan Rows": 106,               +
           "Plan Width": 6,                +
           "Filter": "(c1 < 5)"            +
         }                                 +
       ]                                   +
     }                                     +
   }                                       +
 ]
(1 row)

explain format=Json delete from ed_t where c1 < 5;
                 QUERY PLAN                 
--------------------------------------------
 [                                         +
   {                                       +
     "Plan": {                             +
       "Node Type": "ModifyTable",         +
       "Operation": "Delete",              +
       "Relation Name": "ed_t",            +
       "Alias": "ed_t",                    +
       "Startup Cost": 0.00,               +
       "Total Cost": 13.99,                +
       "Plan Rows": 106,                   +
       "Plan Width": 6,                    +
       "Plans": [                          +
         {                                 +
           "Node Type": "Seq Scan",        +
           "Parent Relationship": "Member",+
           "Relation Name": "ed_t",        +
           "Alias": "ed_t",                +
           "Startup Cost": 0.00,           +
           "Total Cost": 13.99,            +
           "Plan Rows": 106,               +
           "Plan Width": 6,                +
           "Filter": "(c1 < 5)"            +
         }                                 +
       ]                                   +
     }                                     +
   }                                       +
 ]
(1 row)

explain format=JsOn delete from ed_t where c1 < 5;
                 QUERY PLAN                 
--------------------------------------------
 [                                         +
   {                                       +
     "Plan": {                             +
       "Node Type": "ModifyTable",         +
       "Operation": "Delete",              +
       "Relation Name": "ed_t",            +
       "Alias": "ed_t",                    +
       "Startup Cost": 0.00,               +
       "Total Cost": 13.99,                +
       "Plan Rows": 106,                   +
       "Plan Width": 6,                    +
       "Plans": [                          +
         {                                 +
           "Node Type": "Seq Scan",        +
           "Parent Relationship": "Member",+
           "Relation Name": "ed_t",        +
           "Alias": "ed_t",                +
           "Startup Cost": 0.00,           +
           "Total Cost": 13.99,            +
           "Plan Rows": 106,               +
           "Plan Width": 6,                +
           "Filter": "(c1 < 5)"            +
         }                                 +
       ]                                   +
     }                                     +
   }                                       +
 ]
(1 row)

explain format='JsOn' delete from ed_t where c1 < 5;
                 QUERY PLAN                 
--------------------------------------------
 [                                         +
   {                                       +
     "Plan": {                             +
       "Node Type": "ModifyTable",         +
       "Operation": "Delete",              +
       "Relation Name": "ed_t",            +
       "Alias": "ed_t",                    +
       "Startup Cost": 0.00,               +
       "Total Cost": 13.99,                +
       "Plan Rows": 106,                   +
       "Plan Width": 6,                    +
       "Plans": [                          +
         {                                 +
           "Node Type": "Seq Scan",        +
           "Parent Relationship": "Member",+
           "Relation Name": "ed_t",        +
           "Alias": "ed_t",                +
           "Startup Cost": 0.00,           +
           "Total Cost": 13.99,            +
           "Plan Rows": 106,               +
           "Plan Width": 6,                +
           "Filter": "(c1 < 5)"            +
         }                                 +
       ]                                   +
     }                                     +
   }                                       +
 ]
(1 row)

explain format='json' delete from ed_t where c1 < 5;
                 QUERY PLAN                 
--------------------------------------------
 [                                         +
   {                                       +
     "Plan": {                             +
       "Node Type": "ModifyTable",         +
       "Operation": "Delete",              +
       "Relation Name": "ed_t",            +
       "Alias": "ed_t",                    +
       "Startup Cost": 0.00,               +
       "Total Cost": 13.99,                +
       "Plan Rows": 106,                   +
       "Plan Width": 6,                    +
       "Plans": [                          +
         {                                 +
           "Node Type": "Seq Scan",        +
           "Parent Relationship": "Member",+
           "Relation Name": "ed_t",        +
           "Alias": "ed_t",                +
           "Startup Cost": 0.00,           +
           "Total Cost": 13.99,            +
           "Plan Rows": 106,               +
           "Plan Width": 6,                +
           "Filter": "(c1 < 5)"            +
         }                                 +
       ]                                   +
     }                                     +
   }                                       +
 ]
(1 row)

explain format=TRADITIONAL select c1, c2, c3 from ed_t;
                       QUERY PLAN                        
---------------------------------------------------------
 Seq Scan on ed_t  (cost=0.00..13.19 rows=319 width=226)
(1 row)

explain format=TRADITIONAL insert into ed_t values(100, 'hello', 100);
                    QUERY PLAN                    
--------------------------------------------------
 [Bypass]
 Insert on ed_t  (cost=0.00..0.01 rows=1 width=0)
   ->  Result  (cost=0.00..0.01 rows=1 width=0)
(3 rows)

explain format=TRaDITiONAL update ed_t set c2 = 'world' where c1 < 5;
                          QUERY PLAN                          
--------------------------------------------------------------
 Update on ed_t  (cost=0.00..13.99 rows=106 width=14)
   ->  Seq Scan on ed_t  (cost=0.00..13.99 rows=106 width=14)
         Filter: (c1 < 5)
(3 rows)

explain format=traditional delete from ed_t where c1 < 5;
                         QUERY PLAN                          
-------------------------------------------------------------
 Delete on ed_t  (cost=0.00..13.99 rows=106 width=6)
   ->  Seq Scan on ed_t  (cost=0.00..13.99 rows=106 width=6)
         Filter: (c1 < 5)
(3 rows)

explain format='traditional' delete from ed_t where c1 < 5;
                         QUERY PLAN                          
-------------------------------------------------------------
 Delete on ed_t  (cost=0.00..13.99 rows=106 width=6)
   ->  Seq Scan on ed_t  (cost=0.00..13.99 rows=106 width=6)
         Filter: (c1 < 5)
(3 rows)

explain format='TraDitional' delete from ed_t where c1 < 5;
                         QUERY PLAN                          
-------------------------------------------------------------
 Delete on ed_t  (cost=0.00..13.99 rows=106 width=6)
   ->  Seq Scan on ed_t  (cost=0.00..13.99 rows=106 width=6)
         Filter: (c1 < 5)
(3 rows)

drop table ed_t;
drop schema db_explain_desc cascade;
reset current_schema;
